eclipse/src/saros/filesystem/EclipseReferencePointManager.java

```diff
@@ -0,0 +1,215 @@
+
+    return findMember(referencePoint, eclipsePath);
+  }
+
+  /**
+   * Returns a handle to the {@link IFile} file identified by the given {@link IPath} path and
+   * {@link IReferencePoint} reference point
+   *
+   * @param referencePoint The {@link IReferencePoint} reference point, on which the {@link IFile}
+   *     file belongs to
+   * @param referencePointRelativePath the relative {@link IPath} path from the {@link
+   *     IReferencePoint} reference point to the {@link IFile} file
+   * @return a handle to the {@link IFile} file
+   */
+  public IFile getFile(IReferencePoint referencePoint, IPath referencePointRelativePath) {
+    if (referencePoint == null) throw new IllegalArgumentException("Reference point is null");
+
+    if (referencePointRelativePath == null) throw new IllegalArgumentException("Path is null");
+
+    org.eclipse.core.resources.IProject project = getProject(referencePoint);
+
+    if (project == null)
+      throw new NullPointerException(
+          "For reference point " + referencePoint + " does'nt exist a project.");
+
+    return project.getFile(referencePointRelativePath);
+  }
+
+  /**
+   * Returns a handle to the {@link IFile} file represented by this {@link SPath} SPath
+   *
+   * @param path the {@link SPath} SPath which represents the {@link IFile} file
+   * @return a handle to the {@link IFile} file
+   */
+  public IFile getFile(SPath path) {
+    if (path == null) throw new IllegalArgumentException("SPath is null");
+
+    IReferencePoint referencePoint = path.getReferencePoint();
+    saros.filesystem.IPath referencePointRelativePath = path.getProjectRelativePath();
+
+    if (referencePointRelativePath == null)
+      throw new NullPointerException(
+          String.format("SPath %s does not contain a relative path to the file", path));
+
+    IPath eclipsePath = ((EclipsePathImpl) referencePointRelativePath).getDelegate();
+
+    return getFile(referencePoint, eclipsePath);
+  }
+
+  /**
+   * Returns a handle to the {@link IFolder} folder identified by the given {@link IPath} path and
+   * {@link IReferencePoint} reference point
+   *
+   * @param referencePoint The {@link IReferencePoint} reference point, on which the {@link IFolder}
+   *     folder belongs to
+   * @param referencePointRelativePath the {@link IPath} relative path from the {@link
+   *     IReferencePoint} reference point to the {@link IFolder} folder
+   * @return a handle to the {@link IFolder} folder
+   */
+  public IFolder getFolder(IReferencePoint referencePoint, IPath referencePointRelativePath) {
+    if (referencePoint == null) throw new IllegalArgumentException("Reference point is null");
+
+    if (referencePointRelativePath == null) throw new IllegalArgumentException("Path is null");
+
+    org.eclipse.core.resources.IProject project = getProject(referencePoint);
+
+    if (project == null)
+      throw new IllegalArgumentException(
```

Comment:
here you throw a IllegalArgumentException on other places NullPointerException. please stick to one if the logic doesn't differ and javadoc it

Reply:
Ok, I use IllegalArgumentException. For NPE checks, I use Objects.requireNonNull now, like you suggested.